src/hooks/auth.ts

```diff
@@ -65,16 +68,30 @@ export const useAuth = ({ middleware }: { middleware?: any } = {}) => {
   const logout = async () => {
     await api.post('/api/logout');
     mutate(null);
+    destroyCookie(null, 'auth_token'); // Remove the auth token cookie
     router.push('/login');
   };
 
   useEffect(() => {
+    const cookies = parseCookies();
+
     if (user || error) {
       setIsLoading(false);
     }
 
     if (middleware == 'guest' && user) router.push('/campaign');
     if (middleware == 'auth' && error) router.push('/login');
+
+    // Check if auth token cookie exists
+    if (!user && !error && cookies.auth_token) {
+      setAuthToken(cookies.auth_token);
+      mutate();
+    }
+
+    // If user is not logged in and auth token cookie does not exist, redirect to login page
+    if (!cookies.auth_token) {
+      router.push('/login');
+    }
```

Comment:
shouldn't this be checked before everything? If the auth token is not present, then the first action should be to redirect to login route?

Reply:
I think yes. If the user is not logged in then he can access another page of the URL. For this, a check has been made that if the user is not logged in, then the `authToken` of that user will be empty in the cookies. so he can't access any page without a login.

Reply:
No my intention was that this check should be placed at the beginning, before other checks. If this condition is true, then we don't need to check any further and this function code should terminate after redirecting to login.

Reply:
Now, I updated the code accordingly to your comment.